### PR TITLE
Fix the sanity check at the end of the daemon deployment script.

### PR DESCRIPTION
**Pull request by ttung**
1. Upload the required bundle instead of assuming it's already present.
2. Fully parameterize the script so future changes do not require as many changes.

### CODE REVIEW

**Comment by ttung** [comment]
Test:

```
[czipa1osx133 (.venv)]:~/hca/data-store:tonytung-lambda-sanity-check> daemons/invoke_lambda.sh dss-sync dev tests/daemons/sample_s3_bundle_created_event.json tests/daemons/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z
upload: tests/daemons/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z to s3://org-humancellatlas-dss-test/bundles/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z
START RequestId: 2ada7cfb-673c-11e7-89dc-1be8fe6e0448 Version: $LATEST
Domovoi dispatch of event {'Records': [{'eventVersion': '2.0', 'eventSource': 'aws:s3', 'awsRegion': 'us-east-1', 'eventTime': '1970-01-01T00:00:00.000Z', 'eventName': 'ObjectCreated:Put', 'userIdentity': {'principalId': 'AIDAJDPLRKLG7UEXAMPLE'}, 'requestParameters': {'sourceIPAddress': '127.0.0.1'}, 'responseElements': {'x-amz-request-id': 'C3D13FE58DE4C810', 'x-amz-id-2': 'FMyUVURIY8/IgAtTv8xRjskZQpcIZ9KG4V5Wp6S7S/JRWeUWerMUE5JgHvANOjpD'}, 's3': {'s3SchemaVersion': '1.0', 'configurationId': 'testConfigRule', 'bucket': {'name': 'org-humancellatlas-dss-test', 'ownerIdentity': {'principalId': 'A3NL1KOZZKExample'}, 'arn': 'arn:aws:s3:::org-humancellatlas-dss-test'}, 'object': {'key': 'bundles/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z', 'size': 94, 'eTag': 'afbb9e8a3032a50f90b579a0c6a410eb', 'versionId': '096fKKXTRTtl3on89fVO.nfljtsv6qko', 'sequencer': '0055AED6DCD90281E5'}}}]}Domovoi - INFO - Begin transfer of bundles/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z from s3 to gs
Domovoi - INFO - Completed transfer of bundles/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z from org-humancellatlas-dss-test to org-humancellatlas-dss-test
NoneEND RequestId: 2ada7cfb-673c-11e7-89dc-1be8fe6e0448
REPORT RequestId: 2ada7cfb-673c-11e7-89dc-1be8fe6e0448	Duration: 548.19 ms	Billed Duration: 600 ms 	Memory Size: 1536 MB	Max Memory Used: 68 MB	
[czipa1osx133 (.venv)]:~/hca/data-store:tonytung-lambda-sanity-check> echo $?
0
```

File: daemons/invoke_lambda.sh [hunk 1]
```diff
@@ -2,16 +2,25 @@
...
 
-if [[ $# != 3 ]]; then
-    echo "Usage: $(basename $0) daemon-name stage lambda-input-file"
+if [[ $# != 4 ]]; then
+    echo "Usage: $(basename $0) daemon-name stage lambda-input-file bundle-file"
     exit 1
 fi
 
+
+
```
Comment (kislyuk) [inline daemons/invoke_lambda.sh]: Too much whitespace! 2 lines please 😭

File: daemons/Makefile [hunk 2]
```diff
@@ -8,6 +8,6 @@ dss-sync dss-index:
 	cp "$(GOOGLE_APPLICATION_CREDENTIALS)" $@/domovoilib/gcp-credentials.json
 	./build_deploy_config.sh $@ $(STAGE)
 	cd $@; domovoi deploy --stage $(STAGE)
-	./invoke_lambda.sh $@ $(STAGE) ../tests/sample_s3_bundle_created_event.json
+	./invoke_lambda.sh $@ $(STAGE) ../tests/daemons/sample_s3_bundle_created_event.json.template ../tests/daemons/a47b90b2-0967-4fbf-87bc-c6c12db3fedf.2017-07-12T055120.037644Z
```
Comment (kislyuk) [inline daemons/Makefile]: If you're not going to mangle the filename, I guess it's better to call it `blah.template.json` so it still gets recognized as json.

Comment (ttung) [inline daemons/Makefile]: It's no longer valid json.  Does that matter?

Comment (kislyuk) [inline daemons/Makefile]: Oh. In that case, I suppose I don't care either way.

**Comment by ttung** [comment]
re @mikebaumann's comment: i added code to detect the region of a bucket.

**Pull request merged**